File: cpp-tiny/lib/Models/GetMessages.cpp


#include "GetMessages.h"

using namespace Tiny;

GetMessages::GetMessages()
{
	avatar_url = null;
	client = null;
	content = null;
	content_type = null;
	display_recipient = null;
	id = null;
	is_me_message = null;
	reactions = null;
	recipient_id = null;
	sender_email = null;
	sender_full_name = null;
	sender_id = null;
	sender_realm_str = null;
	stream_id = null;
	subject = null;
	topic_links = null;
	submessages = null;
	timestamp = null;
	type = null;
	flags = std::list<std::string>();
	last_edit_timestamp = int(0);
	match_content = std::string();
	match_subject = std::string();
}

GetMessages::GetMessages(std::string jsonString)
{
	this->fromJson(jsonString);
}

GetMessages::~GetMessages()
{

}

void
GetMessages::fromJson(std::string jsonObj)
{   
    bourne::json object = bourne::json::parse(jsonObj);

    const char *avatar_urlKey = "avatar_url";
    
    if(object.has_key(avatar_urlKey)) 
    {
        bourne::json value = object[avatar_urlKey];


        

        AnyType* obj = &avatar_url;
		obj->fromJson(value.dump());

    }

    const char *clientKey = "client";
    
    if(object.has_key(clientKey)) 
    {
        bourne::json value = object[clientKey];


        

        AnyType* obj = &client;
		obj->fromJson(value.dump());

    }

    const char *contentKey = "content";
    
    if(object.has_key(contentKey)) 
    {
        bourne::json value = object[contentKey];


        

        AnyType* obj = &content;
		obj->fromJson(value.dump());

    }

    const char *content_typeKey = "content_type";
    
    if(object.has_key(content_typeKey)) 
    {
        bourne::json value = object[content_typeKey];


        

        AnyType* obj = &content_type;
		obj->fromJson(value.dump());

    }

    const char *display_recipientKey = "display_recipient";
    
    if(object.has_key(display_recipientKey)) 
    {
        bourne::json value = object[display_recipientKey];


        

        AnyType* obj = &display_recipient;
		obj->fromJson(value.dump());

    }

    const char *idKey = "id";
    
    if(object.has_key(idKey)) 
    {
        bourne::json value = object[idKey];


        

        AnyType* obj = &id;
		obj->fromJson(value.dump());

    }

    const char *is_me_messageKey = "is_me_message";
    
    if(object.has_key(is_me_messageKey)) 
    {
        bourne::json value = object[is_me_messageKey];


        

        AnyType* obj = &is_me_message;
		obj->fromJson(value.dump());

    }

    const char *reactionsKey = "reactions";
    
    if(object.has_key(reactionsKey)) 
    {
        bourne::json value = object[reactionsKey];


        

        AnyType* obj = &reactions;
		obj->fromJson(value.dump());

    }

    const char *recipient_idKey = "recipient_id";
    
    if(object.has_key(recipient_idKey)) 
    {
        bourne::json value = object[recipient_idKey];


        

        AnyType* obj = &recipient_id;
		obj->fromJson(value.dump());

    }

    const char *sender_emailKey = "sender_email";
    
    if(object.has_key(sender_emailKey)) 
    {
        bourne::json value = object[sender_emailKey];


        

        AnyType* obj = &sender_email;
		obj->fromJson(value.dump());

    }

    const char *sender_full_nameKey = "sender_full_name";
    
    if(object.has_key(sender_full_nameKey)) 
    {
        bourne::json value = object[sender_full_nameKey];


        

        AnyType* obj = &sender_full_name;
		obj->fromJson(value.dump());

    }

    const char *sender_idKey = "sender_id";
    
    if(object.has_key(sender_idKey)) 
    {
        bourne::json value = object[sender_idKey];


        

        AnyType* obj = &sender_id;
		obj->fromJson(value.dump());

    }

    const char *sender_realm_strKey = "sender_realm_str";
    
    if(object.has_key(sender_realm_strKey)) 
    {
        bourne::json value = object[sender_realm_strKey];


        

        AnyType* obj = &sender_realm_str;
		obj->fromJson(value.dump());

    }

    const char *stream_idKey = "stream_id";
    
    if(object.has_key(stream_idKey)) 
    {
        bourne::json value = object[stream_idKey];


        

        AnyType* obj = &stream_id;
		obj->fromJson(value.dump());

    }

    const char *subjectKey = "subject";
    
    if(object.has_key(subjectKey)) 
    {
        bourne::json value = object[subjectKey];


        

        AnyType* obj = &subject;
		obj->fromJson(value.dump());

    }

    const char *topic_linksKey = "topic_links";
    
    if(object.has_key(topic_linksKey)) 
    {
        bourne::json value = object[topic_linksKey];


        

        AnyType* obj = &topic_links;
		obj->fromJson(value.dump());

    }

    const char *submessagesKey = "submessages";
    
    if(object.has_key(submessagesKey)) 
    {
        bourne::json value = object[submessagesKey];


        

        AnyType* obj = &submessages;
		obj->fromJson(value.dump());

    }

    const char *timestampKey = "timestamp";
    
    if(object.has_key(timestampKey)) 
    {
        bourne::json value = object[timestampKey];


        

        AnyType* obj = &timestamp;
		obj->fromJson(value.dump());

    }

    const char *typeKey = "type";
    
    if(object.has_key(typeKey)) 
    {
        bourne::json value = object[typeKey];


        

        AnyType* obj = &type;
		obj->fromJson(value.dump());

    }

    const char *flagsKey = "flags";
    
    if(object.has_key(flagsKey)) 
    {
        bourne::json value = object[flagsKey];


        std::list<std::string> flags_list;
        std::string element;
        for(auto& var : value.array_range())
        {

            jsonToValue(&element, var, "std::string");
            
            
            flags_list.push_back(element);
        }
        flags = flags_list;


    }

    const char *last_edit_timestampKey = "last_edit_timestamp";
    
    if(object.has_key(last_edit_timestampKey)) 
    {
        bourne::json value = object[last_edit_timestampKey];


        
        jsonToValue(&last_edit_timestamp, value, "int");


    }

    const char *match_contentKey = "match_content";
    
    if(object.has_key(match_contentKey)) 
    {
        bourne::json value = object[match_contentKey];


        
        jsonToValue(&match_content, value, "std::string");


    }

    const char *match_subjectKey = "match_subject";
    
    if(object.has_key(match_subjectKey)) 
    {
        bourne::json value = object[match_subjectKey];


        
        jsonToValue(&match_subject, value, "std::string");


    }


}

bourne::json
GetMessages::toJson()
{
    bourne::json object = bourne::json::object();

    




	object["avatar_url"] = getAvatarUrl().toJson();

    




	object["client"] = getClient().toJson();

    




	object["content"] = getContent().toJson();

    




	object["content_type"] = getContentType().toJson();

    




	object["display_recipient"] = getDisplayRecipient().toJson();

    




	object["id"] = getId().toJson();

    




	object["is_me_message"] = getIsMeMessage().toJson();

    




	object["reactions"] = getReactions().toJson();

    




	object["recipient_id"] = getRecipientId().toJson();

    




	object["sender_email"] = getSenderEmail().toJson();

    




	object["sender_full_name"] = getSenderFullName().toJson();

    




	object["sender_id"] = getSenderId().toJson();

    




	object["sender_realm_str"] = getSenderRealmStr().toJson();

    




	object["stream_id"] = getStreamId().toJson();

    




	object["subject"] = getSubject().toJson();

    




	object["topic_links"] = getTopicLinks().toJson();

    




	object["submessages"] = getSubmessages().toJson();

    




	object["timestamp"] = getTimestamp().toJson();

    




	object["type"] = getType().toJson();

    
    
    
    std::list<std::string> flags_list = getFlags();
    bourne::json flags_arr = bourne::json::array();

    for(auto& var : flags_list)
    {
        flags_arr.append(var);
    }
    object["flags"] = flags_arr;

    



    



    object["last_edit_timestamp"] = getLastEditTimestamp();


    



    object["match_content"] = getMatchContent();


    



    object["match_subject"] = getMatchSubject();



    return object;

}

AnyType
GetMessages::getAvatarUrl()
{
	return avatar_url;
}

void
GetMessages::setAvatarUrl(AnyType  avatar_url)
{
	this->avatar_url = avatar_url;
}

AnyType
GetMessages::getClient()
{
	return client;
}

void
GetMessages::setClient(AnyType  client)
{
	this->client = client;
}

AnyType
GetMessages::getContent()
{
	return content;
}

void
GetMessages::setContent(AnyType  content)
{
	this->content = content;
}

AnyType
GetMessages::getContentType()
{
	return content_type;
}

void
GetMessages::setContentType(AnyType  content_type)
{
	this->content_type = content_type;
}

AnyType
GetMessages::getDisplayRecipient()
{
	return display_recipient;
}

void
GetMessages::setDisplayRecipient(AnyType  display_recipient)
{
	this->display_recipient = display_recipient;
}

AnyType
GetMessages::getId()
{
	return id;
}

void
GetMessages::setId(AnyType  id)
{
	this->id = id;
}

AnyType
GetMessages::getIsMeMessage()
{
	return is_me_message;
}

void
GetMessages::setIsMeMessage(AnyType  is_me_message)
{
	this->is_me_message = is_me_message;
}

AnyType
GetMessages::getReactions()
{
	return reactions;
}

void
GetMessages::setReactions(AnyType  reactions)
{
	this->reactions = reactions;
}

AnyType
GetMessages::getRecipientId()
{
	return recipient_id;
}

void
GetMessages::setRecipientId(AnyType  recipient_id)
{
	this->recipient_id = recipient_id;
}

AnyType
GetMessages::getSenderEmail()
{
	return sender_email;
}

void
GetMessages::setSenderEmail(AnyType  sender_email)
{
	this->sender_email = sender_email;
}

AnyType
GetMessages::getSenderFullName()
{
	return sender_full_name;
}

void
GetMessages::setSenderFullName(AnyType  sender_full_name)
{
	this->sender_full_name = sender_full_name;
}

AnyType
GetMessages::getSenderId()
{
	return sender_id;
}

void
GetMessages::setSenderId(AnyType  sender_id)
{
	this->sender_id = sender_id;
}

AnyType
GetMessages::getSenderRealmStr()
{
	return sender_realm_str;
}

void
GetMessages::setSenderRealmStr(AnyType  sender_realm_str)
{
	this->sender_realm_str = sender_realm_str;
}

AnyType
GetMessages::getStreamId()
{
	return stream_id;
}

void
GetMessages::setStreamId(AnyType  stream_id)
{
	this->stream_id = stream_id;
}

AnyType
GetMessages::getSubject()
{
	return subject;
}

void
GetMessages::setSubject(AnyType  subject)
{
	this->subject = subject;
}

AnyType
GetMessages::getTopicLinks()
{
	return topic_links;
}

void
GetMessages::setTopicLinks(AnyType  topic_links)
{
	this->topic_links = topic_links;
}

AnyType
GetMessages::getSubmessages()
{
	return submessages;
}

void
GetMessages::setSubmessages(AnyType  submessages)
{
	this->submessages = submessages;
}

AnyType
GetMessages::getTimestamp()
{
	return timestamp;
}

void
GetMessages::setTimestamp(AnyType  timestamp)
{
	this->timestamp = timestamp;
}

AnyType
GetMessages::getType()
{
	return type;
}

void
GetMessages::setType(AnyType  type)
{
	this->type = type;
}

std::list<std::string>
GetMessages::getFlags()
{
	return flags;
}

void
GetMessages::setFlags(std::list <std::string> flags)
{
	this->flags = flags;
}

int
GetMessages::getLastEditTimestamp()
{
	return last_edit_timestamp;
}

void
GetMessages::setLastEditTimestamp(int  last_edit_timestamp)
{
	this->last_edit_timestamp = last_edit_timestamp;
}

std::string
GetMessages::getMatchContent()
{
	return match_content;
}

void
GetMessages::setMatchContent(std::string  match_content)
{
	this->match_content = match_content;
}

std::string
GetMessages::getMatchSubject()
{
	return match_subject;
}

void
GetMessages::setMatchSubject(std::string  match_subject)
{
	this->match_subject = match_subject;
}

File: java-play-framework/app/apimodels/GetMessages.java
package apimodels;

import apimodels.GetMessagesAllOf;
import apimodels.MessagesBase;
import java.util.ArrayList;
import java.util.List;
import com.fasterxml.jackson.annotation.*;
import java.util.Set;
import javax.validation.*;
import java.util.Objects;
import javax.validation.constraints.*;
/**
 * GetMessages
 */
@javax.annotation.Generated(value = "org.openapitools.codegen.languages.JavaPlayFrameworkCodegen", date = "2021-08-10T11:44:22.210302Z[Etc/UTC]")
@SuppressWarnings({"UnusedReturnValue", "WeakerAccess"})
public class GetMessages   {
  @JsonProperty("avatar_url")
  
  private Object avatarUrl = null;

  @JsonProperty("client")
  
  private Object client = null;

  @JsonProperty("content")
  
  private Object content = null;

  @JsonProperty("content_type")
  
  private Object contentType = null;

  @JsonProperty("display_recipient")
  
  private Object displayRecipient = null;

  @JsonProperty("id")
  
  private Object id = null;

  @JsonProperty("is_me_message")
  
  private Object isMeMessage = null;

  @JsonProperty("reactions")
  
  private Object reactions = null;

  @JsonProperty("recipient_id")
  
  private Object recipientId = null;

  @JsonProperty("sender_email")
  
  private Object senderEmail = null;

  @JsonProperty("sender_full_name")
  
  private Object senderFullName = null;

  @JsonProperty("sender_id")
  
  private Object senderId = null;

  @JsonProperty("sender_realm_str")
  
  private Object senderRealmStr = null;

  @JsonProperty("stream_id")
  
  private Object streamId = null;

  @JsonProperty("subject")
  
  private Object subject = null;

  @JsonProperty("topic_links")
  
  private Object topicLinks = null;

  @JsonProperty("submessages")
  
  private Object submessages = null;

  @JsonProperty("timestamp")
  
  private Object timestamp = null;

  @JsonProperty("type")
  
  private Object type = null;

  @JsonProperty("flags")
  
  private List<String> flags = null;

  @JsonProperty("last_edit_timestamp")
  
  private Integer lastEditTimestamp;

  @JsonProperty("match_content")
  
  private String matchContent;

  @JsonProperty("match_subject")
  
  private String matchSubject;

  public GetMessages avatarUrl(Object avatarUrl) {
    this.avatarUrl = avatarUrl;
    return this;
  }

   /**
   * Get avatarUrl
   * @return avatarUrl
  **/
  public Object getAvatarUrl() {
    return avatarUrl;
  }

  public void setAvatarUrl(Object avatarUrl) {
    this.avatarUrl = avatarUrl;
  }

  public GetMessages client(Object client) {
    this.client = client;
    return this;
  }

   /**
   * Get client
   * @return client
  **/
  public Object getClient() {
    return client;
  }

  public void setClient(Object client) {
    this.client = client;
  }

  public GetMessages content(Object content) {
    this.content = content;
    return this;
  }

   /**
   * Get content
   * @return content
  **/
  public Object getContent() {
    return content;
  }

  public void setContent(Object content) {
    this.content = content;
  }

  public GetMessages contentType(Object contentType) {
    this.contentType = contentType;
    return this;
  }

   /**
   * Get contentType
   * @return contentType
  **/
  public Object getContentType() {
    return contentType;
  }

  public void setContentType(Object contentType) {
    this.contentType = contentType;
  }

  public GetMessages displayRecipient(Object displayRecipient) {
    this.displayRecipient = displayRecipient;
    return this;
  }

   /**
   * Get displayRecipient
   * @return displayRecipient
  **/
  public Object getDisplayRecipient() {
    return displayRecipient;
  }

  public void setDisplayRecipient(Object displayRecipient) {
    this.displayRecipient = displayRecipient;
  }

  public GetMessages id(Object id) {
    this.id = id;
    return this;
  }

   /**
   * Get id
   * @return id
  **/
  public Object getId() {
    return id;
  }

  public void setId(Object id) {
    this.id = id;
  }

  public GetMessages isMeMessage(Object isMeMessage) {
    this.isMeMessage = isMeMessage;
    return this;
  }

   /**
   * Get isMeMessage
   * @return isMeMessage
  **/
  public Object getIsMeMessage() {
    return isMeMessage;
  }

  public void setIsMeMessage(Object isMeMessage) {
    this.isMeMessage = isMeMessage;
  }

  public GetMessages reactions(Object reactions) {
    this.reactions = reactions;
    return this;
  }

   /**
   * Get reactions
   * @return reactions
  **/
  public Object getReactions() {
    return reactions;
  }

  public void setReactions(Object reactions) {
    this.reactions = reactions;
  }

  public GetMessages recipientId(Object recipientId) {
    this.recipientId = recipientId;
    return this;
  }

   /**
   * Get recipientId
   * @return recipientId
  **/
  public Object getRecipientId() {
    return recipientId;
  }

  public void setRecipientId(Object recipientId) {
    this.recipientId = recipientId;
  }

  public GetMessages senderEmail(Object senderEmail) {
    this.senderEmail = senderEmail;
    return this;
  }

   /**
   * Get senderEmail
   * @return senderEmail
  **/
  public Object getSenderEmail() {
    return senderEmail;
  }

  public void setSenderEmail(Object senderEmail) {
    this.senderEmail = senderEmail;
  }

  public GetMessages senderFullName(Object senderFullName) {
    this.senderFullName = senderFullName;
    return this;
  }

   /**
   * Get senderFullName
   * @return senderFullName
  **/
  public Object getSenderFullName() {
    return senderFullName;
  }

  public void setSenderFullName(Object senderFullName) {
    this.senderFullName = senderFullName;
  }

  public GetMessages senderId(Object senderId) {
    this.senderId = senderId;
    return this;
  }

   /**
   * Get senderId
   * @return senderId
  **/
  public Object getSenderId() {
    return senderId;
  }

  public void setSenderId(Object senderId) {
    this.senderId = senderId;
  }

  public GetMessages senderRealmStr(Object senderRealmStr) {
    this.senderRealmStr = senderRealmStr;
    return this;
  }

   /**
   * Get senderRealmStr
   * @return senderRealmStr
  **/
  public Object getSenderRealmStr() {
    return senderRealmStr;
  }

  public void setSenderRealmStr(Object senderRealmStr) {
    this.senderRealmStr = senderRealmStr;
  }

  public GetMessages streamId(Object streamId) {
    this.streamId = streamId;
    return this;
  }

   /**
   * Get streamId
   * @return streamId
  **/
  public Object getStreamId() {
    return streamId;
  }

  public void setStreamId(Object streamId) {
    this.streamId = streamId;
  }

  public GetMessages subject(Object subject) {
    this.subject = subject;
    return this;
  }

   /**
   * Get subject
   * @return subject
  **/
  public Object getSubject() {
    return subject;
  }

  public void setSubject(Object subject) {
    this.subject = subject;
  }

  public GetMessages topicLinks(Object topicLinks) {
    this.topicLinks = topicLinks;
    return this;
  }

   /**
   * Get topicLinks
   * @return topicLinks
  **/
  public Object getTopicLinks() {
    return topicLinks;
  }

  public void setTopicLinks(Object topicLinks) {
    this.topicLinks = topicLinks;
  }

  public GetMessages submessages(Object submessages) {
    this.submessages = submessages;
    return this;
  }

   /**
   * Get submessages
   * @return submessages
  **/
  public Object getSubmessages() {
    return submessages;
  }

  public void setSubmessages(Object submessages) {
    this.submessages = submessages;
  }

  public GetMessages timestamp(Object timestamp) {
    this.timestamp = timestamp;
    return this;
  }

   /**
   * Get timestamp
   * @return timestamp
  **/
  public Object getTimestamp() {
    return timestamp;
  }

  public void setTimestamp(Object timestamp) {
    this.timestamp = timestamp;
  }

  public GetMessages type(Object type) {
    this.type = type;
    return this;
  }

   /**
   * Get type
   * @return type
  **/
  public Object getType() {
    return type;
  }

  public void setType(Object type) {
    this.type = type;
  }

  public GetMessages flags(List<String> flags) {
    this.flags = flags;
    return this;
  }

  public GetMessages addFlagsItem(String flagsItem) {
    if (flags == null) {
      flags = new ArrayList<>();
    }
    flags.add(flagsItem);
    return this;
  }

   /**
   * The user's [message flags][message-flags] for the message.  [message-flags]: /api/update-message-flags#available-flags 
   * @return flags
  **/
  public List<String> getFlags() {
    return flags;
  }

  public void setFlags(List<String> flags) {
    this.flags = flags;
  }

  public GetMessages lastEditTimestamp(Integer lastEditTimestamp) {
    this.lastEditTimestamp = lastEditTimestamp;
    return this;
  }

   /**
   * The UNIX timestamp for when the message was last edited, in UTC seconds. 
   * @return lastEditTimestamp
  **/
  public Integer getLastEditTimestamp() {
    return lastEditTimestamp;
  }

  public void setLastEditTimestamp(Integer lastEditTimestamp) {
    this.lastEditTimestamp = lastEditTimestamp;
  }

  public GetMessages matchContent(String matchContent) {
    this.matchContent = matchContent;
    return this;
  }

   /**
   * Only present if keyword search was included among the narrow parameters. HTML content of a queried message that matches the narrow, with `<span class=\"highlight\">` elements wrapping the matches for the search keywords. 
   * @return matchContent
  **/
  public String getMatchContent() {
    return matchContent;
  }

  public void setMatchContent(String matchContent) {
    this.matchContent = matchContent;
  }

  public GetMessages matchSubject(String matchSubject) {
    this.matchSubject = matchSubject;
    return this;
  }

   /**
   * Only present if keyword search was included among the narrow parameters. HTML-escaped topic of a queried message that matches the narrow, with `<span class=\"highlight\">` elements wrapping the matches for the search keywords. 
   * @return matchSubject
  **/
  public String getMatchSubject() {
    return matchSubject;
  }

  public void setMatchSubject(String matchSubject) {
    this.matchSubject = matchSubject;
  }


  @Override
  public boolean equals(Object o) {
    if (this == o) {
      return true;
    }
    if (o == null || getClass() != o.getClass()) {
      return false;
    }
    GetMessages getMessages = (GetMessages) o;
    return Objects.equals(avatarUrl, getMessages.avatarUrl) &&
        Objects.equals(client, getMessages.client) &&
        Objects.equals(content, getMessages.content) &&
        Objects.equals(contentType, getMessages.contentType) &&
        Objects.equals(displayRecipient, getMessages.displayRecipient) &&
        Objects.equals(id, getMessages.id) &&
        Objects.equals(isMeMessage, getMessages.isMeMessage) &&
        Objects.equals(reactions, getMessages.reactions) &&
        Objects.equals(recipientId, getMessages.recipientId) &&
        Objects.equals(senderEmail, getMessages.senderEmail) &&
        Objects.equals(senderFullName, getMessages.senderFullName) &&
        Objects.equals(senderId, getMessages.senderId) &&
        Objects.equals(senderRealmStr, getMessages.senderRealmStr) &&
        Objects.equals(streamId, getMessages.streamId) &&
        Objects.equals(subject, getMessages.subject) &&
        Objects.equals(topicLinks, getMessages.topicLinks) &&
        Objects.equals(submessages, getMessages.submessages) &&
        Objects.equals(timestamp, getMessages.timestamp) &&
        Objects.equals(type, getMessages.type) &&
        Objects.equals(flags, getMessages.flags) &&
        Objects.equals(lastEditTimestamp, getMessages.lastEditTimestamp) &&
        Objects.equals(matchContent, getMessages.matchContent) &&
        Objects.equals(matchSubject, getMessages.matchSubject);
  }

  @Override
  public int hashCode() {
    return Objects.hash(avatarUrl, client, content, contentType, displayRecipient, id, isMeMessage, reactions, recipientId, senderEmail, senderFullName, senderId, senderRealmStr, streamId, subject, topicLinks, submessages, timestamp, type, flags, lastEditTimestamp, matchContent, matchSubject);
  }

  @SuppressWarnings("StringBufferReplaceableByString")
  @Override
  public String toString() {
    StringBuilder sb = new StringBuilder();
    sb.append("class GetMessages {\n");
    
    sb.append("    avatarUrl: ").append(toIndentedString(avatarUrl)).append("\n");
    sb.append("    client: ").append(toIndentedString(client)).append("\n");
    sb.append("    content: ").append(toIndentedString(content)).append("\n");
    sb.append("    contentType: ").append(toIndentedString(contentType)).append("\n");
    sb.append("    displayRecipient: ").append(toIndentedString(displayRecipient)).append("\n");
    sb.append("    id: ").append(toIndentedString(id)).append("\n");
    sb.append("    isMeMessage: ").append(toIndentedString(isMeMessage)).append("\n");
    sb.append("    reactions: ").append(toIndentedString(reactions)).append("\n");
    sb.append("    recipientId: ").append(toIndentedString(recipientId)).append("\n");
    sb.append("    senderEmail: ").append(toIndentedString(senderEmail)).append("\n");
    sb.append("    senderFullName: ").append(toIndentedString(senderFullName)).append("\n");
    sb.append("    senderId: ").append(toIndentedString(senderId)).append("\n");
    sb.append("    senderRealmStr: ").append(toIndentedString(senderRealmStr)).append("\n");
    sb.append("    streamId: ").append(toIndentedString(streamId)).append("\n");
    sb.append("    subject: ").append(toIndentedString(subject)).append("\n");
    sb.append("    topicLinks: ").append(toIndentedString(topicLinks)).append("\n");
    sb.append("    submessages: ").append(toIndentedString(submessages)).append("\n");
    sb.append("    timestamp: ").append(toIndentedString(timestamp)).append("\n");
    sb.append("    type: ").append(toIndentedString(type)).append("\n");
    sb.append("    flags: ").append(toIndentedString(flags)).append("\n");
    sb.append("    lastEditTimestamp: ").append(toIndentedString(lastEditTimestamp)).append("\n");
    sb.append("    matchContent: ").append(toIndentedString(matchContent)).append("\n");
    sb.append("    matchSubject: ").append(toIndentedString(matchSubject)).append("\n");
    sb.append("}");
    return sb.toString();
  }

  /**
   * Convert the given object to string with each line indented by 4 spaces
   * (except the first line).
   */
  private String toIndentedString(Object o) {
    if (o == null) {
      return "null";
    }
    return o.toString().replace("\n", "\n    ");
  }
}

